lib/Formatters/EventzFormatter.php

```diff
@@ -101,22 +101,38 @@ public function format( array $layout ) : array {
      * Create recurring events as single item.
      *
      * @param array $events Events.
+     * @param array $query_params Query parameters.
      *
      * @return void
      */
-    public static function create_recurring_events( $events )  {
+    public static function create_recurring_events( $events, $query_params )  {
 
         $recurring_events = [];
         if( ! empty( $events['events'] ) ) {
             foreach ( $events['events'] as $event ) {
+
+                // Chek if event has dates or entries
                 if ( count( $event['dates'] ) > 1 ) {
-                    foreach ( $event['dates'] as $date ) {
+                    $recurring_event_dates = $event['dates'];
+                } else if ( ! empty( $event['entries'] ) ) {
+                    $recurring_event_dates = $event['entries'];
+                }
+
+                // Get recurring event single dates
+                if ( isset( $recurring_event_dates ) ) {
```

Comment:
Tekisin niin päin että initialisoisin ko. variablen castatulla array:lla ja tarkistaisin onko se tyhjä niin saat samalla kiinni jos $event['dates'] esim. on tyhjä

```suggestion
// Aiemmin ennen iffejä
$recurring_event_dates = [];
...

                if ( ! empty( $recurring_event_dates ) ) {
```